libraries/Microsoft.Bot.Builder.Core/Adapters/TestAdapter.cs

```diff
@@ -73,17 +73,18 @@ public new TestAdapter Use(IMiddleware middleware)
 
         public async override Task<ResourceResponse[]> SendActivities(ITurnContext context, Activity[] activities)
         {
-            List<ResourceResponse> responses = new List<ResourceResponse>();
+            var responses = new ResourceResponse[activities.Length];
 
-            foreach (var activity in activities)
+            for(var index = 0; index < activities.Length; index++)
```

Comment:
Please add a comment here (and in the other spots) describing this is a traditional for loop rather than a for-each. Reviewers of this code a year from now won't have the context, and current best practices call for for-each in place of for loops.

Reply:
✔️